Bottom: 12a3f5b01b6dd1bcdebf98dd1af90b1d118893cd
Top:    5e206bba9048567b1f7c0f28d066e5a8b8ee60cf
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-05 13:58:16 +0800

autofs-5.0.7 - fix spawn_umount() return check in mount_bind.c:lookup_init()

Check return of spawn_umount() and report it if it fails.


---

diff --git a/modules/mount_bind.c b/modules/mount_bind.c
index 4975294..d6c6fe7 100644
--- a/modules/mount_bind.c
+++ b/modules/mount_bind.c
@@ -57,7 +57,8 @@ int mount_init(void **context)
 		bind_works = 1;
 	}
 
-	spawn_umount(LOGOPT_NONE, "-n", t2_dir, NULL);
+	if (spawn_umount(LOGOPT_NONE, "-n", t2_dir, NULL) != 0)
+		debug(LOGOPT_ANY, MODPREFIX "umount failed for %s", t2_dir);
 
 out:
 	rmdir(t1_dir);
